app/lib/features/space/settings/pages/visibility_accessibility_page.dart

```diff
@@ -174,13 +221,25 @@ class _VisibilityAccessibilityPageState
     );
   }
 
-  void removeSpace(SpaceItem spaceItem) {
+  Future<void> removeSpace(Space? space, SpaceItem spaceItem) async {
+    final room =
+        await ref.read(maybeRoomProvider(space!.getRoomIdStr()).future);
+    if (room != null) {
+      if (await space.isChildSpaceOf(spaceItem.roomId)) {
+        await room.removeParentRoom(spaceItem.roomId, null);
```

Comment:
So addition and removal of spaces for Limited permission is all about adding child and removing child?